app/src/main/java/org/fossasia/susi/ai/skills/skillSearch/SearchSkillFragment.kt

```diff
@@ -0,0 +1,139 @@
+package org.fossasia.susi.ai.skills.skillSearch
+
+import android.content.Context
+import android.os.Bundle
+import android.support.annotation.NonNull
+import android.support.v4.app.Fragment
+import android.support.v4.widget.SwipeRefreshLayout
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.SnapHelper
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.fragment_group_wise_skill_listing.*
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.dataclasses.GroupWiseSkills
+import org.fossasia.susi.ai.helper.SimpleDividerItemDecoration
+import org.fossasia.susi.ai.helper.StartSnapHelper
+import org.fossasia.susi.ai.rest.responses.susi.SkillData
+import org.fossasia.susi.ai.skills.SkillFragmentCallback
+import org.fossasia.susi.ai.skills.groupwiseskills.adapters.recycleradapters.SkillsListAdapter
+import org.fossasia.susi.ai.skills.groupwiseskills.contract.IGroupWiseSkillsPresenter
+import org.fossasia.susi.ai.skills.groupwiseskills.contract.IGroupWiseSkillsView
+import timber.log.Timber
+
+/**
+ *
+ * Created by naman653 on 28/03/2019.
+ */
+class SearchSkillFragment : Fragment(), IGroupWiseSkillsView, SwipeRefreshLayout.OnRefreshListener {
+
+    private lateinit var query: String
+    private lateinit var skillSet: List<SkillData>
+    private lateinit var skillAdapterSnapHelper: SnapHelper
+    private lateinit var searchSkillPresenter: IGroupWiseSkillsPresenter
+    private lateinit var skills: GroupWiseSkills
+    private lateinit var skillsAdapter: SkillsListAdapter
+    private lateinit var skillCallback: SkillFragmentCallback
+
+    companion object {
+        const val SEARCH_KEY = "search_key"
+        const val SKILL_DATA = "skill_data"
+        fun newInstance(skillData: ArrayList<SkillData>, searchQuery: String): SearchSkillFragment {
+            val fragment = SearchSkillFragment()
+            val bundle = Bundle()
+            bundle.putParcelableArrayList(SKILL_DATA, skillData)
+            bundle.putString(SEARCH_KEY, searchQuery)
+            fragment.arguments = bundle
+
+            return fragment
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return inflater.inflate(R.layout.fragment_group_wise_skill_listing, container, false)
+    }
+
+    @NonNull
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        query = arguments!!.getString(SEARCH_KEY)
+        skillSet = arguments!!.getParcelableArrayList(SKILL_DATA)
```

Comment:
Don't use !! operator.

Reply:
Well, I was unable to do that because it creates a problem in casting. I tried to remove that but the base declarations are not coping with the change. I'll try but if you can suggest something. @iamareebjamal Please suggest any alternative approach.

Reply:
You can use ?. operator and check. I hope it would work.

Reply:
Of course, I'd tried that, but as I said it creates a problem in casting. Maybe @iamareebjamal can suggest a better approach

Reply:
Sorry, I resolved that with using ?. operator. Thanks for your review.